src/pages/Productdetail/Productdetail.js

```diff
@@ -1,7 +1,172 @@
-import './Productdetail.scss';
+import './ProductDetail.scss';
+import { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { getProducts } from '../../service/tetcher';
 
-function Productdetail() {
-  return <div>Productdetail</div>;
-}
+const ProductDetail = ({ convertPrice, cart, setCart }) => {
+  const { id } = useParams();
+  const [product, setProduct] = useState({});
+  const [count, setCount] = useState(1);
 
-export default Productdetail;
+  const navigate = useNavigate();
+
+  const handleQuantity = type => {
+    if (type === 'plus') {
+      setCount(count + 1);
+    } else {
+      if (count === 1) return;
+      setCount(count - 1);
+    }
+  };
+  useEffect(() => {
+    getProducts().then(data => {
+      setProduct(
+        data.data.products.find(product => product.id === parseInt(id))
+      );
+    });
+  }, [id]);
+
+  const setQuantity = (id, quantity) => {
+    const found = cart.filter(el => el.id === id)[0];
+    const idx = cart.indexOf(found);
+    const cartItem = {
+      id: product.id,
+      image: product.image,
+      name: product.name,
+      price: product.price,
+      provider: product.provider,
+      quantity: quantity,
+      explanation: product.explanation,
+    };
+    setCart([...cart.slice(0, idx), cartItem, ...cart.slice(idx + 1)]);
+  };
+
+  const handleCart = () => {
+    const cartItem = {
+      id: product.id,
+      image: product.image,
+      name: product.name,
+      price: product.price,
+      provider: product.provider,
+      quantity: count,
+      explanation: product.explanation,
+    };
+    const found = cart.find(el => el.id === cartItem.id);
+    if (found) setQuantity(cartItem.id, found.quantity + count);
+    else {
+      setCart([...cart, cartItem]);
+    }
+  };
+
+  return (
+    product && (
+      <div id="renew">
```

Comment:
그리고, 컴포넌트의 최상위 태그는 컴포넌트 이름과 동일하게 쓰는게 위코드 컨벤션입니다.
```suggestion
      <div className="productDetail">
```